upload_vehicle_data.py

```diff
@@ -0,0 +1,327 @@
+
+# Setup logging
+logging.basicConfig(
+    filename='upload_vehicle_data.log',
+    level=logging.INFO,
+    format='%(asctime)s %(levelname)s: %(message)s',
+)
+
+
+def run_ssh_command(remote_user, remote_ip, password, command):
+    """Run a command on the remote machine using SSH."""
+    ssh_command = (
+        f"sshpass -p {password} ssh {remote_user}@{remote_ip} '{command}'"
+    )
+    try:
+        subprocess.run(ssh_command, shell=True, check=True)
+        logging.info(f'Ran SSH command: {command}')
+    except subprocess.CalledProcessError as e:
+        logging.error(f'Failed to run SSH command: {command}: {e}')
+        raise
+
+
+def create_remote_directory(
+    remote_user, remote_ip, remote_temp_directory, password
+):
+    """Create a directory on the remote machine."""
+    command = f'mkdir -p {remote_temp_directory}'
+    run_ssh_command(remote_user, remote_ip, password, command)
+
+
+def delete_remote_directory(
+    remote_user, remote_ip, remote_temp_directory, password
+):
+    """Delete a directory on the remote machine."""
+    command = f'rm -rf {remote_temp_directory}'
+    run_ssh_command(remote_user, remote_ip, password, command)
+
+
+def compress_and_transfer_rosbag(
+    remote_user,
+    remote_ip,
+    rosbag_path,
+    remote_temp_directory,
+    vdi_upload_directory,
+    password,
+):
+    """Compress rosbag on remote machine and transfer it to VDI machine."""
+    # Compress the rosbag on the remote machine
+    remote_compressed_path = os.path.join(
+        remote_temp_directory, os.path.basename(rosbag_path)
+    )
+    mcap_path = '/home/linuxbrew/.linuxbrew/bin/mcap'
+    compress_cmd = (
+        f'{mcap_path} compress {rosbag_path} -o {remote_compressed_path}'
+    )
+    try:
+        run_ssh_command(remote_user, remote_ip, password, compress_cmd)
+        logging.info(
+            f'Compressed rosbag {rosbag_path} to'
+            f'{remote_compressed_path} on the remote machine.'
+        )
+    except subprocess.CalledProcessError as e:
+        logging.error(
+            f'Failed to compress rosbag {rosbag_path} on remote machine: {e}'
+        )
+        return False
+
+    # Transfer the compressed file to the VDI machine
+    rsync_cmd = [
+        'sshpass',
+        '-p',
+        password,
+        'rsync',
+        '-avz',
+        f'{remote_user}@{remote_ip}:{remote_compressed_path}',
+        vdi_upload_directory,
+    ]
+    try:
+        subprocess.run(rsync_cmd, check=True)
+        logging.info(
+            f'Transferred compressed rosbag {remote_compressed_path}'
+            f'to the VDI machine at {vdi_upload_directory}.'
+        )
+    except subprocess.CalledProcessError as e:
+        logging.error(
+            f'Failed to transfer compressed rosbag'
+            f'{remote_compressed_path} from remote machine: {e}'
+        )
+        return False
+
+    # Remove the compressed file from the remote machine
+    remove_remote_file_cmd = f'rm {remote_compressed_path}'

```

Comment:
How do we validate that the transfered bag is not corrupted? I believe `rsync` can do that, could you please give it a check?

Reply:
added checksum option.